lib/models/activity.dart

```diff
@@ -2,126 +2,173 @@ import 'package:connect_plus/models/activityDate.dart';
+  String get sId => _sId;
+  DateTime get startDate => _startDate;
+  DateTime get endDate => _endDate;
+  String get onBehalfOf => _onBehalfOf;
+  String get venue => _venue;
+  String get zoomID => _zoomID;
+  String get days => _days;
+  DateTime get createdAt => _createdAt;
+  DateTime get updatedAt => _updatedAt;
+  int get iV => _iV;
+  AdminUser get createdBy => _createdBy;
+  AdminUser get updatedBy => _updatedBy;
+  String get id => _id;
+  ERG get erg => _erg;
+  List<ActivityDate> get activityDates => _activityDates;
 
-    zoomID = json['zoomID'];
-    days = json['days'];
-
-    recurrenceDates = json['recurrenceDates'] != null
-        ? json['recurrenceDates'].cast<DateTime>()
-        : null;
+  Activity({
+    @required String recurrence,
+    @required ImageFile poster,
+    @required String sId,
+    @required String name,
+    @required String onBehalfOf,
+    @required DateTime startDate,
+    @required DateTime endDate,
+    @required String zoomID,
+    @required String venue,
+    @required String days,
+    @required DateTime createdAt,
+    @required DateTime updatedAt,
+    @required int iV,
+    @required this.recurrenceDates,
+    @required AdminUser createdBy,
+    @required AdminUser updatedBy,
+    @required String id,
+    @required ERG erg,
+    @required List<ActivityDate> activityDates,
+    @required bool slider,
+  })  : _recurrence = recurrence,
+        _poster = poster,
+        _sId = sId,
+        _name = name,
+        _onBehalfOf = onBehalfOf,
+        _startDate = startDate,
+        _endDate = endDate,
+        _zoomID = zoomID,
+        _venue = venue,
+        _days = days,
+        _createdAt = createdAt,
+        _updatedAt = updatedAt,
+        _iV = iV,
+        _createdBy = createdBy,
+        _updatedBy = updatedBy,
+        _id = id,
+        _erg = erg,
+        _activityDates = activityDates,
+        _slider = slider;
 
-    endDate =
-        json['endDate'] != null ? DateTime.parse((json['endDate'])).toLocal() : null;
-    startDate =
-        json['startDate'] != null ? DateTime.parse((json['startDate'])).toLocal() : null;
-    createdAt =
-        json['createdAt'] != null ? DateTime.parse((json['createdAt'])) : null;
-    updatedAt =
-        json['updatedAt'] != null ? DateTime.parse((json['updatedAt'])) : null;
-    venue = json['venue'];
-    iV = json['__v'];
-    createdBy = json['created_by'] != null
-        ? new AdminUser.fromJson(json['created_by'])
-        : null;
-    updatedBy = json['updated_by'] != null
-        ? new AdminUser.fromJson(json['updated_by'])
-        : null;
-    erg = json['erg'] != null ? new ERG.fromJson(json['erg']) : null;
-    if (json['activity_dates'] != null) {
-      activityDates = new List<ActivityDate>();
-      json['activity_dates'].forEach((v) {
-        activityDates.add(new ActivityDate.fromJson(v));
-      });
-    }
-    id = json['id'];
+  factory Activity.fromJson(Map<String, dynamic> json) {
+    return Activity(
```

Comment:
I refactored the `fromJson` methods of these classes because its cleaner, and will prevent other contributors from forgetting to set a field.